core/src/main/java/com/novoda/noplayer/mediaplayer/forwarder/CompletionInfoForwarder.java

```diff
@@ -0,0 +1,24 @@
+package com.novoda.noplayer.mediaplayer.forwarder;
+
+import android.media.MediaPlayer;
+
+import com.novoda.noplayer.listeners.InfoListeners;
+
+import java.util.HashMap;
+
+class CompletionInfoForwarder implements MediaPlayer.OnCompletionListener {
+
+    private final InfoListeners infoListeners;
+
+    public CompletionInfoForwarder(InfoListeners infoListeners) {
+        this.infoListeners = infoListeners;
+    }
+
+    @Override
+    public void onCompletion(MediaPlayer mp) {
+        HashMap<String, String> keyValuePairs = new HashMap<>();
+        keyValuePairs.put("mp", String.valueOf(mp));
```

Comment:
Could change this but then it will stop working for the primitives that also come through on the interfaces. I would rather leave as a `String` because at the end of the day it is expected to be used for logging.